tests/instruments/ametek/test_ametek7270.py

```diff
@@ -0,0 +1,480 @@
+     7),
+    ([(b'REFN 7', b'')],
+     7),
+    ([(b'REFN 7', b'')],
+     7),
+    ([(b'REFN 7', b'')],
+     7),
+    ([(b'REFN 7', b'')],
+     7),
+))
+def test_harmonic_setter(comm_pairs, value):
+    with expected_protocol(
+            Ametek7270,
+            comm_pairs,
+    ) as inst:
+        inst.harmonic = value
+
+
+def test_phase_setter():
+    with expected_protocol(
+            Ametek7270,
+            [(b'REFP. 250', b'')],
+    ) as inst:
+        inst.phase = 250
+
+
+def test_voltage_setter():
+    with expected_protocol(
+            Ametek7270,
+            [(b'OA. 2.7', b'')],
+    ) as inst:
+        inst.voltage = 2.7
+
+
+def test_frequency_setter():
+    with expected_protocol(
+            Ametek7270,
+            [(b'OF. 12000', b'')],
+    ) as inst:
+        inst.frequency = 12000
+
+
+def test_dac1_setter():
+    with expected_protocol(
+            Ametek7270,
+            [(b'DAC. 1 7', b'')],
+    ) as inst:
+        inst.dac1 = 7
+
+
+def test_dac2_setter():
+    with expected_protocol(
+            Ametek7270,
+            [(b'DAC. 2 -7', b'')],
+    ) as inst:
+        inst.dac2 = -7
+
+
+def test_dac3_setter():
+    with expected_protocol(
+            Ametek7270,
+            [(b'DAC. 3 2.6', b'')],
+    ) as inst:
+        inst.dac3 = 2.6
+
+
+def test_dac4_setter():
+    with expected_protocol(
+            Ametek7270,
+            [(b'DAC. 4 5.5', b'')],
+    ) as inst:
+        inst.dac4 = 5.5
+
+
+@pytest.mark.parametrize("comm_pairs, value", (
+    ([],
+     True),
+    ([],
+     False),
+))
+def test_autogain_setter(comm_pairs, value):
+    with expected_protocol(
+            Ametek7270,
+            comm_pairs,
+    ) as inst:
+        inst.autogain = value
+
+
+def test_set_voltage_mode():
+    with expected_protocol(
+            Ametek7270,
+            [(b'IMODE 0', b'')],
+    ) as inst:
+        assert inst.set_voltage_mode() is None
+
+
+def test_set_differential_mode():
+    with expected_protocol(
+            Ametek7270,
+            [(b'VMODE 3', b''),
+             (b'LF 3 0', b'')],
+    ) as inst:
+        assert inst.set_differential_mode() is None
+
+
+@pytest.mark.parametrize("comm_pairs, args, kwargs, value", (
+    ([(b'IMODE 1', b'')],
+     (), {'low_noise': False}, None),
+    ([(b'IMODE 2', b'')],
+     (), {'low_noise': True}, None),
+))
+def test_set_current_mode(comm_pairs, args, kwargs, value):
+    with expected_protocol(
+            Ametek7270,
+            comm_pairs,
+    ) as inst:
+        assert inst.set_current_mode(*args, **kwargs) == value
+
+
+def test_set_channel_A_mode():
+    with expected_protocol(
+            Ametek7270,
+            [(b'VMODE 1', b'')],
+    ) as inst:
+        assert inst.set_channel_A_mode() is None
+
+
+def test_id():
+    with expected_protocol(
+            Ametek7270,
+            [(b'ID', b'7270\n'),
+             (b'VER', b'2.11\n')],
+    ) as inst:
+        assert inst.identification() == '7270/2.11'
```

Comment:
You forgot to change that:
```suggestion
        assert inst.id == '7270/2.11'
```

Reply:
indeed, you have a keen eye! That's done, thanks for the extended review!